### PR TITLE
feat: Support cloning environment variables when cloning an environment

### PR DESCRIPTION
**Pull request by shanewxy**
**Problem:**

Support cloning environment variables when cloning an environment.

**Solution:**

- Modify API to support variables inputs when creating environments.
- Add variable validation and creation logic 

**Related Issue:**
#985

### CODE REVIEW

**Review by rainfd — DISMISSED** [review]
LGTM

File: pkg/apis/environment/basic.go [hunk 1]
```diff
@@ -46,6 +46,24 @@ func (h Handler) Create(req CreateRequest) (CreateResponse, error) {
...
+		}
+
+		variables, err := tx.Variables().CreateBulk().
+			Set(variableInputs...).
+			Save(req.Context)
+		if err != nil {
+			return err
+		}
+
+		entity.Edges.Variables = variables
```
Comment (alexcodelf) [inline pkg/apis/environment/basic.go:65]: Can you use a function `EnvironmentVariablesEdgeSave` like `dao.EnvironmentConnectorsEdgeSave` and use it in SaveE?

Comment (shanewxy) [reply pkg/apis/environment/basic.go]: Moved variables save logic to dao. Moving service creation to dao will cause import cycle, so leave it for now.

**Review by alexcodelf — APPROVED** [review]
LGTM

**Pull request merged**